### PR TITLE
(GH-240) TCP Retry defaults

### PR DESCRIPTION
**Pull request by jpogran**
This implements tcp retry defaults for the languageclient

- [x] A delay between retry
- [x] A max attempt count
- [x] An actual UI Warning then Error message when it cannot bind a port

### CODE REVIEW

**Comment by jpogran** [comment]
Added UI warning message.

image

**Comment by jpogran** [comment]
Error colllapsed:
image

Error expanded:
image

File: src/connection.ts [hunk 1]
```diff
@@ -228,19 +228,65 @@ export class ConnectionManager implements IConnectionManager {
...
+              }
+            );
+
+            return null;
+          }else{
+            attempt = attempt + 1;
+            var message = `Timed out connecting to language server. Is the server running at ${address}:${port} ? Will wait timeout value before trying again`;
+            switch(err.code){
+              case 'ETIMEDOUT':
+                message = `Timed out connecting to language server. Is the server running at ${address}:${port} ? Will wait timeout value before trying again`;
```
Comment (glennsarti) [inline src/connection.ts]: Should `Will wait timeout value before trying again` be `Will wait timeout 5 seconds before trying again`

Comment (jpogran) [reply src/connection.ts]: The timeout for ETIMEOUT is ~30 seconds. I didn’t see a method to change that

**Pull request merged**